src/averageAges.js

```diff
@@ -49,13 +76,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * If `onlyWithSon` is specified then function calculates age difference only
  * for sons and their mothers.
  *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
+//  * @param {object[]} people
+//  * @param {boolean} onlyWithSon - optional
+//  *
+//  * @return {number}
+//  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const differences = [];
+
+  people.forEach(person => {
+    const personChildren = onlyWithSon
+      ? people
+        .filter(child => person.name === child.mother && child.sex === 'm')
+      : people
+        .filter(child => person.name === child.mother);
+
+    const ageDifference = personChildren
+      .map(child => child.born - person.born);
+
+    differences
+      .push(...ageDifference);
+  });
+
+  const avgDifference = differences
+    .reduce((sum, current) => sum + current, 0)
+    / differences.length;
```

Comment:
You do the same calculations with reduce in every function. Maybe we should move this calculation to a separate helper?